src/main/java/org/jenkinsci/plugins/vstest_runner/VsTestInstallation.java

```diff
@@ -61,8 +68,63 @@ protected Object readResolve() {
         return this;
     }
 
+    public String getVsTestExe() {
+        return getHome();
+    }
+
+    public static VsTestInstallation getDefaultInstallation() {
+        DescriptorImpl vstestsTools = Jenkins.getInstance().getDescriptorByType(VsTestInstallation.DescriptorImpl.class);
+        VsTestInstallation tool = vstestsTools.getInstallation(VsTestInstallation.DEFAULT);
+        if (tool != null) {
+            return tool;
+        } else {
+            VsTestInstallation[] installations = vstestsTools.getInstallations();
+            if (installations.length > 0) {
+                return installations[0];
+            } else {
+                onLoaded();
+                return vstestsTools.getInstallations()[0];
+            }
+        }
+    }
+
+    @Initializer(after = InitMilestone.EXTENSIONS_AUGMENTED)
+    public static void onLoaded() {
+        DescriptorImpl descriptor = (DescriptorImpl) Jenkins.getInstance().getDescriptor(VsTestInstallation.class);
+        VsTestInstallation[] installations = getInstallations(descriptor);
+
+        if (installations != null && installations.length > 0) {
+            // No need to initialize if there's already something
+            return;
+        }
+
+
+        String defaultVSTestExe = isWindows() ? "vstest.console.exe" : "vstest.console";
+        VsTestInstallation tool = new VsTestInstallation(DEFAULT, defaultVSTestExe);
+        descriptor.setInstallations(tool);
+        descriptor.save();
+    }
+
+    private static VsTestInstallation[] getInstallations(DescriptorImpl descriptor) {
+        VsTestInstallation[] installations = null;
+        try {
+            installations = descriptor.getInstallations();
+        } catch (NullPointerException e) {
+            installations = new VsTestInstallation[0];
+        }
+        return installations;
+    }
+
+    @Override
+    public DescriptorImpl getDescriptor() {
+        return (DescriptorImpl) Jenkins.getInstance().getDescriptorOrDie(getClass());
+    }
+
+    private static boolean isWindows() {
+        return File.pathSeparatorChar == ';';

```

Comment:
This is meant for Jenkins Master to be able to default whether ".exe" or without. So if the Jenkins master is running Linux